analytics.go

```diff
@@ -250,25 +251,25 @@ func (c *client) upload(b []byte) error {
 	}
 
 	defer res.Body.Close()
-	c.report(res)
-
-	return nil
+	return c.report(res)
 }
 
 // Report on response body.
-func (c *client) report(res *http.Response) {
+func (c *client) report(res *http.Response) (err error) {
+	var body []byte
+
 	if res.StatusCode < 400 {
```

Comment:
OK.

My only concern is the http client should be handling 3xx automatically already, so getting one at this level kind of feels like a bug / error and we likely shouldn't accept this as a valid response.

How about changing it to `res.StatusCode < 300`?